File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/go/F10.P2.go
package main

import "fmt"

func probe() float64 {
	var i int32 = 3
	var d float64 = 0.5
	// BEGIN PROBE F10.P2
	sum := float64(i) + d
	// END PROBE F10.P2
	return sum
}

func main() {
	fmt.Println(probe())
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/zig/F10.P2.zig
const std = @import("std");

var src: i32 = 3;

fn probe() f64 {
    const i: i32 = src;
    const d: f64 = 0.5;
// BEGIN PROBE F10.P2
const sum = @as(f64, @floatFromInt(i)) + d;
// END PROBE F10.P2
    return sum;
}

pub fn main() !void {
    var threaded: std.Io.Threaded = .init_single_threaded;
    const io = threaded.io();
    var out_buf: [64]u8 = undefined;
    var out = std.Io.File.stdout().writer(io, &out_buf);
    try out.interface.print("{d}\n", .{probe()});
    try out.interface.flush();
}
